modelci/app/v1/endpoints/model.py

```diff
@@ -33,3 +33,31 @@ def get_all_model(name: str = None, framework: Framework_ = None, engine: Engine
 def get_model(*, id: str):  # noqa
     model = ModelService.get_model_by_id(id)
     return ModelDetailOut.from_bo(model)
+
+
+@router.get('/structure/{id}')  # TODO: add response_model
+async def get_model_structure(id: str):  # noqa
+    """
+    Get model structure as a DAG.
+
+    Arguments:
+        id (str): Model object ID.
+    """
+    # return model DAG
+    raise NotImplementedError('Method `get_model_structure` not implemented.')
+
+
+@router.patch('/structure/{id}')  # TODO: add response_model
+def update_model_structure_as_new(id: str, dry_run: bool = False):  # noqa
+    """
+    Update model structure and save as a new version.
```

Comment:
please use more specific comments

Updating model structure is too general. For example, adding a layer to the original model belongs to this definition. However, we do not support this because it may break the whole model.  

I suppose we only need to fine_tune_model_as_new(). The function should accept some parameters like model id, layer name, etc.

Reply:
Currently, I leave this function name, but it only supports editing the last layer. I add a TODO for supporting more model structure edition.